pkg/chunked/storage_linux.go

```diff
@@ -1544,10 +1544,10 @@ func (c *chunkedDiffer) ApplyDiff(dest string, options *archive.TarOptions, diff
 			}
 		}
 
-		r.Name = filepath.Clean(r.Name)
+		r.Name = internal.CleanAbsPath(r.Name)
```

Comment:
One thing I didn't quite understand here is why these ones need to be absolute paths?

Reply:
They don’t need to be, this is a very concise way to eliminate leading `../` entries (at the syntax level, which matters for `*at()` system calls; something still needs to care about symlinks trying to break out).

Reply:
> this is a very concise way to eliminate leading ../ entries (at the syntax level, which matters for *at() system calls; something still needs to care about symlinks trying to break out).

Hmm, only if we're not using `RESOLVE_IN_ROOT` though...which, digging through we are, at least if it's supported.

Anyways, marking this conversation as resolved.

Reply:
`RESOLVE_IN_ROOT` helps for creating a single file; we may need to create all parent directories.

Reply:
AFAIK with RESOLVE_IN_ROOT, `../` is just going to be the same as `/` in that root, so any leading/unnecessary `../` is harmless.

But anyways either way I don't think it matters, this sub-thread is just for building up our own shared understanding.

(On that topic, currently. https://github.com/containers/composefs-rs builds up a userspace in-memory VFS that represents the target composefs metadata file, so the semantics of the kernel openat() as it applies to potentially hostile or just weird user space paths don't come into play)